### PR TITLE
GitHub Action - pypi publish on new release

### PR DESCRIPTION
**Pull request by lucagiove**
Create pythonpublish.yml

### CODE REVIEW

**Comment by lucagiove** [comment]
cannot work I've no permissions for actions on github

**Comment by bulkan** [comment]
@lucagiove if you need help setting this I can.

Also I checked to see why you don't have access to Github Actions and can't seem to work it. 

I could try adding you back as admin?

**Comment by lucagiove** [comment]
Looking at github documentation I need to see the settings button where I could add credentials for pypi.
Thanks!

**Comment by bulkan** [comment]
Sorry I tried removing and adding as collaborator again to see if I could add you as admin :(

**Comment by lucagiove** [comment]
by the way you could add your pypi credentials under Settings > Secrets in order to have this working:
```
        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
        TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
```
never tried but it's the official github action to publish on pypi it should work

**Comment by bulkan** [comment]
Can i use a PyPI token instead ?

**Comment by lucagiove** [comment]
Seems twine doesn't support it..

```
twine upload --help | egrep -i "USER|PASS|TOKEN"
                    [-s] [--sign-with SIGN_WITH] [-i IDENTITY] [-u USERNAME]
                    [-p PASSWORD] [-c COMMENT] [--config-file CONFIG_FILE]
  -u USERNAME, --username USERNAME
                        The username to authenticate to the repository
                        TWINE_USERNAME environment variable.)
  -p PASSWORD, --password PASSWORD
                        The password to authenticate to the repository
                        TWINE_PASSWORD environment variable.)
```

**Comment by bulkan** [comment]
@lucagiove I've added these secrets. I'm using `TWINE_USERNAME=__token__` and `TWINE_PASSWORD=api-key`. This is according to https://github.com/pypa/twine/issues/482#issuecomment-515690438

**Comment by lucagiove** [comment]
Cool! Thanks let's try it :)

File: .github/workflows/pythonpublish.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      with:
+        python-version: '3.x'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install setuptools wheel twine
+    - name: Build and publish
+      env:
+        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
+        TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
```
Comment (bulkan) [inline .github/workflows/pythonpublish.yml]: These secrets are named the same ;

```
   TWINE_USERNAME: ${{ secrets.TWINE_USERNAME }}
   TWINE_PASSWORD: ${{ secrets.TWINE_PASSWORD }}
```

Comment (lucagiove) [reply .github/workflows/pythonpublish.yml]: Actually make sense, I didn't notice just copied and pasted 😁 let me change

**Pull request merged**